sdk/core/Azure.Core/src/Shared/PrivilegedOperationAttribute.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Azure.Core
+{
+    /// <summary>
+    /// Decorates a potentially privileged operation.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Method)]
+    internal class PrivilegedOperationAttribute : Attribute
```

Comment:
It would be helpful to build out the end-to-end story around this attribute a little more prior to moving it into Azure.Core.  Will it be used outside of Storage libraries?  What does it enabled for customers of the libraries?  Depending on the problem it's solving, would it make sense to use something that's already provided in .NET, for example, the [SecurityCriticalAttribute](https://learn.microsoft.com/en-us/dotnet/api/system.security.securitycriticalattribute?view=net-7.0)?  If not, how is this different?

Reply:
Yes - it'll be used across the SDK.  We're starting with Storage, but we'll also be using this in Tables, Identity, and KeyVault in the near future.  This is for internal Azure tooling to audit service implementations with static code analysis.  It's an internal attribute so it won't show up in docs or be obvious to customers who aren't probing with private reflection.  We're also not shipping this in Azure.Core directly but rather using shared source to pull it into projects as needed.  I don't think reusing an existing attribute would work because it'll already be applied to other contexts that may not be applicable for the analysis the Azure security folks are conducting.

Reply:
Thanks, Ted!

One more question - in the case that we move forward with this in Azure.Core, is there value in making it a public type to better address e2e scenarios, and to have it go through arch board review?

Reply:
Is this analysis specific to the Azure SDK packages or part of a larger audit within Microsoft?  Is this applicable to the other languages as well?

Reply:
I don't think it's worth making this a public type.  I have asked the Core Azure folks to setup an Arch Board review to help clarify some of the other questions and process.